forest/benchmarking/noise_model_generation.py

```diff
@@ -0,0 +1,77 @@
+from tqdm import tqdm
+from forest.benchmarking.readout import estimate_joint_confusion_in_set
+from forest.benchmarking.direct_fidelity_estimation import \
+    generate_exhaustive_process_dfe_experiment, acquire_dfe_data, estimate_dfe
+from forest.benchmarking.randomized_benchmarking import average_gate_error_to_rb_decay
+from pyquil import Program
+from pyquil.api import QuantumComputer, get_benchmarker
+from pyquil.noise import NoiseModel, KrausModel, pauli_kraus_map, combine_kraus_maps
+from pyquil.device import gates_in_isa
+from pyquil.gate_matrices import QUANTUM_GATES
+
+
+def _do_dfe(qc: QuantumComputer, program: Program, num_shots: int = 10_000):
+    bm = get_benchmarker()
+    dfe_expt = generate_exhaustive_process_dfe_experiment(program, list(program.get_qubits()), bm)
+    dfe_results = acquire_dfe_data(qc, dfe_expt, num_shots=num_shots)
+    return estimate_dfe(dfe_results, 'process')[0]
+
+
+def _depolarizing_channel(fidelity: float, dim: int):
+    p = average_gate_error_to_rb_decay(1 - fidelity, dim)
+    pr_id = p + (1 - p) / dim
+    pr_other = (1 - pr_id) / (dim ** 2 - 1)
+    probabilities = [pr_id] + [pr_other] * (dim ** 2 - 1)
+    return pauli_kraus_map(probabilities)
+
+
+def generate_depolarizing_noise_model(qc: QuantumComputer, show_progress_bar: bool = True,
```

Comment:
Is it reasonably sufficient (and/or fairly standard) to only consider a depolarizing noise model? What about things like cross-talk?